src/hotspot/cpu/x86/stubGenerator_x86_64.cpp

```diff
@@ -5682,6 +5682,671 @@ address generate_avx_ghash_processBlocks() {
+
+      __ movl(r15, 0x00011000);
+      __ evpbroadcastd(pack32_op, r15, Assembler::AVX_512bit);
+
+      __ cmpl(length, 0xff);
+      __ jcc(Assembler::lessEqual, L_process64);
+
+      // load masks required for decoding data
+      __ BIND(L_processdata);
+      __ evmovdquq(join01, ExternalAddress(StubRoutines::x86::base64_vbmi_join_0_1_addr()), Assembler::AVX_512bit,r13);
+      __ evmovdquq(join12, ExternalAddress(StubRoutines::x86::base64_vbmi_join_1_2_addr()), Assembler::AVX_512bit, r13);
+      __ evmovdquq(join23, ExternalAddress(StubRoutines::x86::base64_vbmi_join_2_3_addr()), Assembler::AVX_512bit, r13);
+
+      __ evmovdquq(tmp16_op3, pack16_op, Assembler::AVX_512bit);
+      __ evmovdquq(tmp16_op2, pack16_op, Assembler::AVX_512bit);
+      __ evmovdquq(tmp16_op1, pack16_op, Assembler::AVX_512bit);
+      __ evmovdquq(tmp32_op3, pack32_op, Assembler::AVX_512bit);
+      __ evmovdquq(tmp32_op2, pack32_op, Assembler::AVX_512bit);
+      __ evmovdquq(tmp32_op1, pack32_op, Assembler::AVX_512bit);
+
+      __ align(32);
+      __ BIND(L_process256);
+      // Grab input data
+      __ evmovdquq(input0, Address(source, start_offset, Address::times_1, 0x00), Assembler::AVX_512bit);
+      __ evmovdquq(input1, Address(source, start_offset, Address::times_1, 0x40), Assembler::AVX_512bit);
+      __ evmovdquq(input2, Address(source, start_offset, Address::times_1, 0x80), Assembler::AVX_512bit);
+      __ evmovdquq(input3, Address(source, start_offset, Address::times_1, 0xc0), Assembler::AVX_512bit);
+
+      // Copy the low part of the lookup table into the destination of the permutation
+      __ evmovdquq(translated0, lookup_lo, Assembler::AVX_512bit);
+      __ evmovdquq(translated1, lookup_lo, Assembler::AVX_512bit);
+      __ evmovdquq(translated2, lookup_lo, Assembler::AVX_512bit);
+      __ evmovdquq(translated3, lookup_lo, Assembler::AVX_512bit);
+
+      // Translate the base64 input into "decoded" bytes
+      __ evpermt2b(translated0, input0, lookup_hi, Assembler::AVX_512bit);
+      __ evpermt2b(translated1, input1, lookup_hi, Assembler::AVX_512bit);
+      __ evpermt2b(translated2, input2, lookup_hi, Assembler::AVX_512bit);
+      __ evpermt2b(translated3, input3, lookup_hi, Assembler::AVX_512bit);
+
+      // OR all of the translations together to check for errors (high-order bit of byte set)
+      __ vpternlogd(t0, 0xfe, input1, input2, Assembler::AVX_512bit);
+
+      __ vpternlogd(t1, 0xfe, translated0, translated1, Assembler::AVX_512bit);
+      __ evmovdquq(t2, translated2, Assembler::AVX_512bit);
+      __ vpternlogd(t2, 0xfe, translated3, t0, Assembler::AVX_512bit);
+      __ evmovdquq(errorvec, t0, Assembler::AVX_512bit);
+      __ vpternlogd(errorvec, 0xfe, t1, t2, Assembler::AVX_512bit);
+
+      // Check if there was an error - if so, try 64-byte chunks
+      __ evpmovb2m(k3, errorvec, Assembler::AVX_512bit);
+      __ kortestql(k3, k3);
+      __ vpxor(errorvec, errorvec, errorvec, Assembler::AVX_512bit);
```

Comment:
Why clearing errorvec is needed here? Seems to be not necessary for the 256 byte processing loop.